src/components/cards/ClientTaskCard.tsx

```diff
@@ -124,12 +128,20 @@ export const ClientTaskCard = ({
                     overflow: 'hidden',
                     fontSize: '12px',
                     color: (theme) => theme.color.gray[500],
+
+                    '@media (max-width: 400px)': {
+                      maxWidth: task.dueDate ? '80px' : '150px',
+                    },
+                    '@media (min-width: 400px) and (max-width: 500px)': {
+                      maxWidth: task.dueDate ? '120px' : '220px',
+                    },
+                    '@media (min-width: 500px) and (max-width: 600px)': {
+                      maxWidth: task.dueDate ? '200px' : '300px',
+                    },
                   }}
```

Comment:
The current implementation of multiple breakpoints for maxWidth in the Typography component seems to be overly complex and potentially difficult to maintain. To reduce the complexity, consider simplifying the breakpoints or using a utility function that calculates maxWidth based on the screen size and task.dueDate condition. This approach could make the code more readable and maintainable while still achieving the desired responsiveness.

I also think we don't need to specify both max-width and min-width here. We can just give max-width and rest will be taken care. Some of the breakpoints seems not be used.